yt_dlp/extractor/common.py

```diff
@@ -860,8 +862,14 @@ def _request_webpage(self, url_or_request, video_id, note=None, errnote=None, fa
             headers = (headers or {}).copy()
             headers.setdefault('X-Forwarded-For', self._x_forwarded_for_ip)
 
+        extensions = None
+        if impersonate is not None:
+            target = ImpersonateTarget.from_str(':' if impersonate is True else impersonate)
```

Comment:
```suggestion
            target = ImpersonateTarget.from_str('' if impersonate is True else impersonate)
```

is more standard I would think